hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3131,3 +3156,142 @@ func reconcileMachineApproverDeployment(deployment *appsv1.Deployment, hc *hyper
 	hyperutil.SetDefaultPriorityClass(deployment)
 	return nil
 }
+
+const oidcDocumentsFinalizer = "hypershift.io/aws-oidc-discovery"
+
+func oidcPublicDocumentPaths() []string {
+	return []string{"/.well-known/openid-configuration", "/openid/v1/jwks"}
+}
+
+func (r *HostedClusterReconciler) reconcileAWSOIDCDocuments(ctx context.Context, hcluster *hyperv1.HostedCluster, hcp *hyperv1.HostedControlPlane) error {
+	if hcp.Status.KubeConfig == nil || hcluster.Spec.Platform.AWS == nil {
+		return nil
+	}
+
+	// We use the presence of the finalizer to short-circuit the document upload to avoid
+	// constantly re-uploading it.
+	if sets.NewString(hcluster.Finalizers...).Has(oidcDocumentsFinalizer) {
+		return nil
+	}
+
+	if r.AWSOIDCBucketName == "" || r.S3Client == nil {
+		return errors.New("hypershift wasn't configured with a S3 bucket or credentials, this makes it unable to set up OIDC for AWS clusters. Please install hypershift with the --aws-oidc-bucket-name, --aws-oidc-bucket-region and --aws-creds flags set. The bucket must pre-exist and the credentials must be authorized to write into it")
```

Comment:
This needs to be added as a `Prerequisites` so the Readme

Reply:
I updated the readme with instructions on the bucket and how `hypershift install` needs to be parameterized now